kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/EngineServiceDiscovery.scala

```diff
@@ -52,3 +61,59 @@ class EngineServiceDiscovery private(
     super.stop()
   }
 }
+
+object EngineServiceDiscovery extends Logging {
+
+  val mapper: ObjectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
+
+  def getEngineByPolicy(
+      zkClient: CuratorFramework,
+      namespace: String,
+      providePolicy: ProvidePolicy): Option[(String, Int)] = {
+    providePolicy match {
+      case RANDOM =>
+        Random.shuffle(getServiceNodesInfo(zkClient, namespace, silent = true)).take(1) match {
+          case Seq(sn) => Some((sn.host, sn.port))
+          case _ => None
+        }
+      case _ => throw new IllegalArgumentException(s"Not support provide policy $providePolicy")
```

Comment:
Seems not required to add a default branch for enum, pls remove it and see if the compiler print any warnings